### PR TITLE
pkg/acirenderer: fix renderer to render empty rootfs directory as well.

### PR DESCRIPTION
**Pull request by yifan-gu**
Related to https://github.com/coreos/rkt/issues/1034#issuecomment-111578112

### CODE REVIEW

File: pkg/acirenderer/acirenderer.go [hunk 1]
```diff
@@ -145,8 +145,9 @@ func getACIFiles(img Image, ap ACIProvider, allFiles map[string]byte, pwlm map[s
 		name := hdr.Name
 		cleanName := filepath.Clean(name)
 
-		// Ignore files outside /rootfs/ (at the moment only "manifest")
-		if !strings.HasPrefix(cleanName, "rootfs/") {
+		// Ignore files outside /rootfs/ (at the moment only "manifest"),
+		// but still make sure we render the rootfs directory, even it's empty.
+		if !strings.HasPrefix(cleanName, "rootfs/") && !(cleanName == "rootfs" && hdr.Typeflag == tar.TypeDir) {
```
Comment (jonboulle) [inline pkg/acirenderer/acirenderer.go]: I don't understand, don't you want to add this to allFiles? am I dumb? help..

Comment (yifan-gu) [inline pkg/acirenderer/acirenderer.go]: @jonboulle I can see that... Replaced with `isRootFsDir()`

**Comment by jonboulle** [comment]
But isn't this going to cause issues with the pathWhitelist check?

**Comment by yifan-gu** [comment]
@jonboulle Fixed.

File: pkg/acirenderer/acirenderer.go [hunk 2]
```diff
@@ -145,7 +145,15 @@ func getACIFiles(img Image, ap ACIProvider, allFiles map[string]byte, pwlm map[s
...
-		// Ignore files outside /rootfs/ (at the moment only "manifest")
+		// Add the rootfs directory.
+		if cleanName == "rootfs" && hdr.Typeflag == tar.TypeDir {
+			ra.FileMap[cleanName] = struct{}{}
+			allFiles[cleanName] = hdr.Typeflag
+			return nil
+		}
+
+		// Ignore files outside /rootfs/ (at the moment only "manifest"),
+		// but keep the rootfs directory, even it's empty.
```
Comment (jonboulle) [inline pkg/acirenderer/acirenderer.go]: Remove this line now

**Comment by jonboulle** [comment]
@yifan-gu why didn't I ask you for a test here? ;-)

**Comment by yifan-gu** [comment]
will do

**Comment by yifan-gu** [comment]
@jonboulle Tests added

**Comment by jonboulle** [comment]
THanks @yifan-gu

**Pull request merged**